lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -321,6 +338,28 @@ walletKeyIsReencrypted (wid, wname) (xprv, pwd) newPwd =
         [ TxOut (Address "destination") (Coin 14) ]
         []
 
+walletListTransactionsSorted
+    :: (WalletId, WalletName)
+    -> Map (Hash "Tx") (Tx, TxMeta)
+    -> Property
+walletListTransactionsSorted (wid, wname) history =
+    monadicIO $ liftIO $ do
+        let state = Map.insert (Address "source") minBound mempty
+        let wallet = (wid, wname, DummyState state)
+        (WalletLayerFixture db wl _) <- liftIO $ setupFixture wallet
+        _ <- runExceptT
+            $ withExceptT (error . show)
+            $ putTxHistory db (PrimaryKey wid) history
+
+        (Right txs) <- runExceptT $ listTransactions wl wid
+        length txs `shouldBe` Map.size history
+        txs `shouldBe` L.sortBy (comparing $ Down . slotId . snd) txs
+        case txs of
+            (a:b:_) ->
+                ((slotId . snd $ a) >= (slotId . snd $ b)) `shouldBe` True
```

Comment:
I'm not sure about this. 

I like the `>=` because it's easy to comprehend, but it is just comparing the first two elements, so we need the re-implementation of the sorting on L354.

Reply:
suggestion:

```hs
isSortedDesc = all (uncurry (<=)) . (\xs -> zip (drop 1 xs) xs) $ txs
if isSortedDesc 
    then return ()
    else $ expectationFailure $ "..."
```

Reply:
I like the `txs == sort txs` form of assertion.